src/main/java/com/android/volley/CacheDispatcher.java

```diff
@@ -134,11 +138,12 @@ public void run() {
 
                     // Post the intermediate response back to the user and have
                     // the delivery then forward the request along to the network.
-                    mDelivery.postResponse(request, response, new Runnable() {
+                    final Request<?> finalRequest = request;
+                    mDelivery.postResponse(finalRequest, response, new Runnable() {
```

Comment:
If I understand correctly, the cause of the bug is that Dalvik won't GC a local variable of a loop even after it goes out of scope on the next iteration of that loop.

If so, by making a final variable here (that is this never nulled out), isn't this just introducing another instance of the same leak?